### PR TITLE
 Fixed passing null is deprecated for htmlspecialchars in Mage_Core_Helper_Abstract

### PR DESCRIPTION
**Pull request by kyrena**
### Description

This fix `passing null to parameter 1 of type string is deprecated` for `htmlspecialchars` with PHP 8.1/8.2.
Perhaps it's not the best way (I never had this warning when `$addSlashes=true`).
It seems that this appear on backend customer edit page.

See also this [discussion comment](https://github.com/OpenMage/magento-lts/discussions/3025#discussioncomment-5195497).

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

**Comment by kyrena** [comment]
lool found the reason: some module `return null` instead of `return ''` for `getTabTitle`...
so perhaps this is useless.

File: app/code/core/Mage/Core/Helper/Abstract.php [hunk 1]
```diff
@@ -343,6 +343,7 @@ public function jsQuoteEscape($data, $quote = '\'')
      */
     public function quoteEscape($data, $addSlashes = false)
     {
+        $data = (string)$data;
```
Comment (kiatng) [inline app/code/core/Mage/Core/Helper/Abstract.php:346]: Is this more efficient?
```suggestion
        if (!$data) {
            return $data;
        }
```

Comment (kyrena) [reply app/code/core/Mage/Core/Helper/Abstract.php:346]: Yes, but this is due to a third party code not up to date, so I close this one?

Comment (kiatng) [reply app/code/core/Mage/Core/Helper/Abstract.php:346]: I think we still need to fix it.

**Comment by fballiano** [comment]
I've commited @kiatng's version, I think it should be ok right?

**Review by fballiano — APPROVED** [review]

File: app/code/core/Mage/Core/Helper/Abstract.php (hunk 1 above)
Comment (fballiano) [reply app/code/core/Mage/Core/Helper/Abstract.php]: I think we should close this one. the function is clearly marked to accept a string and there seems there are no misuse of the type inside our code

**Pull request merged**